src/main/java/minijava/ir/assembler/block/AssemblerFile.java

```diff
@@ -0,0 +1,126 @@
+package minijava.ir.assembler.block;
+
+import com.sun.jna.Platform;
+import java.util.*;
+import minijava.ir.NameMangler;
+import minijava.ir.assembler.GNUAssemblerConvertible;
+import org.jetbrains.annotations.NotNull;
+
+/** List of segments that are combined to an assembler file */
+public class AssemblerFile implements GNUAssemblerConvertible, Collection<Segment> {
```

Comment:
A static method could do that equally well, I'd say. But whatever, this isn't really world changing.